### PR TITLE
Refactoring code module Tags

### PR DESCRIPTION
**Pull request by DimaKacer**
- Implementation code of Directive <tags-list />

### CODE REVIEW

File: angular-spa-ui/js/admin-app/admin-app-srv.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+"use strict";
```
Comment (tarhunakau) [inline angular-spa-ui/js/admin-app/admin-app-srv.js:1]: Do you really need an empty file?

File: angular-spa-ui/js/admin-app/admin-app.js [hunk 2]
```diff
@@ -1,2 +1,25 @@
 "use strict";
-console.log('admin-app');
+
+// Dependency
+var angular = require('angular');
+var uiRouter = require('angular-ui-router');
+var uiBs = require('angular-ui-bootstrap');
```
Comment (tarhunakau) [inline angular-spa-ui/js/admin-app/admin-app.js]: rename it, names must be obvious

**Comment by tarhunakau** [comment]
Fix minor issues

File: angular-spa-ui/js/search-app/tags-module/tags-list-directive/tags-list-ctrl.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    $scope.form = {
+        tagName: ''
+    };
+
+    $scope.msgWorning = '';
+
+    // Define scope directive
+    //-------------------------
+
+    $scope.getCurrentScopeParam = function() {
```
Comment (ValeryShvyndzikau) [inline angular-spa-ui/js/search-app/tags-module/tags-list-directive/tags-list-ctrl.js]: the first one thing: better to have if (typeof $scope.inputParam === 'undefined')

Comment (ValeryShvyndzikau) [inline angular-spa-ui/js/search-app/tags-module/tags-list-directive/tags-list-ctrl.js]: the second: I think it's a not so good approach for switching something, better to have passed options which can say where you use something

File: angular-spa-ui/js/search-app/tags-module/tags-list-directive/tags-list-ctrl.js [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+        if ($scope.inputParam === undefined) {
+            return 'tagsListForTagsPage';
```
Comment (ValeryShvyndzikau) [inline angular-spa-ui/js/search-app/tags-module/tags-list-directive/tags-list-ctrl.js]: also it's so long "tagsListForTagsPage"

Comment (ValeryShvyndzikau) [inline angular-spa-ui/js/search-app/tags-module/tags-list-directive/tags-list-ctrl.js]: and why 'tagsListForTagsPage' -> "FOR" and then 'tagsListByArticleId' BY ?

File: angular-spa-ui/js/search-app/tags-module/tags-list-directive/tags-list-ctrl.js [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+            );
+    };
+
+
+    // Directive of module article
+    //----------------------------
+
+    // Get item article
+    $scope.getArticleItem = function(articleItem) {
+        console.log($scope);
```
Comment (ValeryShvyndzikau) [inline angular-spa-ui/js/search-app/tags-module/tags-list-directive/tags-list-ctrl.js]: Should not appearing in pull requests

File: angular-spa-ui/js/search-app/tags-module/tags-list-directive/tags-list-ctrl.js [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+                    $scope.tags = _.without($scope.tags, itemTag);
+                },
+                function(errorMessage) {
+                    console.warn(errorMessage);
+                }
+            );
+    };
+
+    // Detach item tag from article
+    $scope.detachTagFromArticle = function(itemTag, articleItem) {
```
Comment (ValeryShvyndzikau) [inline angular-spa-ui/js/search-app/tags-module/tags-list-directive/tags-list-ctrl.js]: "Detach" normally useful for unbinding events processes, my be remove would be better

File: angular-spa-ui/js/search-app/tags-module/tags-list-directive/tags-list-ctrl.js [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+
+    // Create or add new tag of current scope
+    //----------------------------
+
+    // Add new tag
+    $scope.addTag = function() {
+        var tagName = $scope.form.tagName;
+
+        if (tagName.length !== 0) {
+            if ($scope.currentScope === 'tagsListForTagsPage') {
```
Comment (ValeryShvyndzikau) [inline angular-spa-ui/js/search-app/tags-module/tags-list-directive/tags-list-ctrl.js]: Copypaste!!!

File: angular-spa-ui/js/search-app/tags-module/tags-list-directive/tags-list-ctrl.js [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+    $scope.getCurrentScope = function() {
+        $scope.currentScope = $scope.getCurrentScopeParam();
+        $scope.getTagsList();
+    };
+
+    // Show tgs list
+    //-------------------------
+
+    $scope.getTagsList = function() {
+        if ($scope.currentScope === 'tagsListForTagsPage') {
```
Comment (ValeryShvyndzikau) [inline angular-spa-ui/js/search-app/tags-module/tags-list-directive/tags-list-ctrl.js]: Copypaste!!!

File: angular-spa-ui/js/search-app/tags-module/tags-list-directive/tags-list-ctrl.js [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+                }
+            );
+    };
+
+    // Remove or detach the current tag of current scope
+    //----------------------------
+
+    // Remove tag
+    $scope.removeTagItem = function(itemTag) {
+        if ($scope.currentScope === 'tagsListForTagsPage') {
```
Comment (ValeryShvyndzikau) [inline angular-spa-ui/js/search-app/tags-module/tags-list-directive/tags-list-ctrl.js]: Copypaste!!!

**Comment by DimaKacer** [comment]
Refactoring code module tags and directive <tags-list/>

File: angular-spa-ui/js/admin-app/admin-app-srv.js (hunk 1 above)
Comment (tarhunakau) [inline angular-spa-ui/js/admin-app/admin-app-srv.js:1]: I do not see changes for that

File: angular-spa-ui/js/admin-app/admin-app.js (hunk 2 above)
Comment (tarhunakau) [inline angular-spa-ui/js/admin-app/admin-app.js]: I do not see changes for that

**Comment by DimaKacer** [comment]
Fix all

**Comment by tarhunakau** [comment]
Could be merged

**Pull request merged**